includes/model/class-blog-user.php

```diff
@@ -0,0 +1,212 @@
+<?php
+namespace Activitypub\Model;
+
+use WP_Query;
+use Activitypub\Signature;
+use Activitypub\Collection\Users;
+
+use function Activitypub\is_user_disabled;
+
+class Blog_User extends User {
+	/**
+	 * The User-ID
+	 *
+	 * @var int
+	 */
+	protected $_id = Users::BLOG_USER_ID; // phpcs:ignore PSR2.Classes.PropertyDeclaration.Underscore
+
+	/**
+	 * The User-Type
+	 *
+	 * @var string
+	 */
+	protected $type = 'Group';
+
+	public static function from_wp_user( $user_id ) {
+		if ( is_user_disabled( $user_id ) ) {
+			return new WP_Error(
+				'activitypub_user_not_found',
+				\__( 'User not found', 'activitypub' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		$object = new static();
+		$object->_id = $user_id;
+
+		return $object;
+	}
+
+	/**
+	 * Get the User-Name.
+	 *
+	 * @return string The User-Name.
+	 */
+	public function get_name() {
+		return \esc_html( \get_bloginfo( 'name' ) );
+	}
+
+	/**
+	 * Get the User-Description.
+	 *
+	 * @return string The User-Description.
+	 */
+	public function get_summary() {
+		return \wpautop(
+			\wp_kses(
+				\get_bloginfo( 'description' ),
+				'default'
+			)
+		);
+	}
+
+	/**
+	 * Get the User-Url.
+	 *
+	 * @return string The User-Url.
+	 */
+	public function get_url() {
+		return \esc_url( \trailingslashit( get_home_url() ) . '@' . $this->get_preferred_username() );
+	}
+
+	/**
+	 * Generate and save a default Username.
+	 *
+	 * @return string The auto-generated Username.
+	 */
+	public static function get_default_username() {
+		$username = \get_option( 'activitypub_blog_user_identifier' );
+
+		if ( $username ) {
+			return $username;
+		}
+
+		// check if domain host has a subdomain
+		$host       = \wp_parse_url( \get_home_url(), \PHP_URL_HOST );
+		$host       = \str_replace( 'www.', '', $host );
```

Comment:
I think it would be better to explicitly use `preg_replace( '/^www\.', '', $host )` here? People put `www` in the strangest-possible places :)